### PR TITLE
Add support for offsetRequestV1 messages

### PR DESCRIPTION
**Pull request by jlafaye**
Those can be used to request the offset of a stream at a given timepoint

### CODE REVIEW

File: kafka/protocol/offset.py [hunk 1]
```diff
@@ -40,6 +53,24 @@ class OffsetRequest_v0(Struct):
...
+    )
+    DEFAULTS = {
+        'replica_id': -1
+    }
+
 
 OffsetRequest = [OffsetRequest_v0]
 OffsetResponse = [OffsetResponse_v0]
+OffsetRequest = [OffsetRequest_v0, OffsetRequest_v1]
+OffsetResponse = [OffsetResponse_v0, OffsetResponse_v1]
```
Comment (dpkp) [inline kafka/protocol/offset.py]: Should only assign the variables once here -- you can delete the earlier lists.

**Comment by jlafaye** [comment]
Hello,

thank you for taking the time to review the pull request. This is my first time so do not hesitate to show me what's wrong so I do not keep doing the wrong things.
I fixed the obvious error in first commit  (a026051).
I noticed that Travis was complaining for one specific version. I could not find anything wrong in the logs. I guess it is a random runtime issue. Should I investigate further ?

File: kafka/protocol/offset.py [hunk 2]
```diff
@@ -40,6 +53,22 @@ class OffsetRequest_v0(Struct):
...
+    API_KEY = 2
+    API_VERSION = 1
+    RESPONSE_TYPE = OffsetResponse_v1
+    SCHEMA = Schema(
+        ('replica_id', Int32),
+        ('topics', Array(
+            ('topic', String('utf-8')),
+            ('partitions', Array(
+                ('partition', Int32),
+                ('time', Int64)))))
```
Comment (dpkp) [inline kafka/protocol/offset.py]: can we call this 'timestamp' to be consistent with the protocol docs ?

**Comment by dpkp** [comment]
Travis test suite is not 100% reliable. In most cases it seems like our fixtures (kafka broker or zookeeper) crash or become unresponsive. You do not need to investigate these errors; I'll typically just restart the job.

**Comment by dpkp** [comment]
thanks!

File: kafka/protocol/offset.py (hunk 2 above)
Comment (jeffwidman) [inline kafka/protocol/offset.py]: See also https://github.com/dpkp/kafka-python/pull/969

**Pull request merged**